ddtrace/appsec/_remoteconfiguration.py

```diff
@@ -26,9 +33,46 @@ def enable_appsec_rc():
 
     if _appsec_rc_features_is_enabled():
         from ddtrace.internal.remoteconfig import RemoteConfig
+        from ddtrace.internal.remoteconfig.constants import ASM_DATA_PRODUCT
         from ddtrace.internal.remoteconfig.constants import ASM_FEATURES_PRODUCT
 
         RemoteConfig.register(ASM_FEATURES_PRODUCT, appsec_rc_reload_features(tracer))
+        RemoteConfig.register(ASM_DATA_PRODUCT, appsec_rc_reload_features(tracer))
+
+
+def _appsec_rules_data(tracer, features):
+    # type: (Tracer, Union[Literal[False], Mapping[str, Any]]) -> None
+    if features and tracer._appsec_processor:
+        rules_data = features.get("rules_data", [])
+        if rules_data:
+            log.debug("Reloading Appsec rules data: %s", rules_data)
+            tracer._appsec_processor.update_rules(rules_data)
+
+
+def _appsec_1click_actication(tracer, features):
+    # type: (Tracer, Union[Literal[False], Mapping[str, Any]]) -> None
+    if features is False:
+        rc_appsec_enabled = False
+    else:
+        rc_appsec_enabled = features.get("asm", {}).get("enabled")
+
+    if rc_appsec_enabled is not None:
+        from ddtrace.internal.remoteconfig import RemoteConfig
+        from ddtrace.internal.remoteconfig.constants import ASM_DATA_PRODUCT
+
+        log.debug("Reloading Appsec 1-click: %s", rc_appsec_enabled)
+        _appsec_enabled = True
+
+        if not (APPSEC_ENV not in os.environ and rc_appsec_enabled is True) and (
```

Comment:
this condition is quite hard to read, can we simplify it a bit since these are booleans

```suggestion
        if not (APPSEC_ENV not in os.environ and rc_appsec_enabled) and (
```